src/ring/adapter/jetty9/common.clj

```diff
@@ -46,3 +46,48 @@
   ;; need to upgrade to websockets protocol.
   (and status (== 101 status) ws))
 
+(defn get-charset [^Request request]
+  (when-let [content-type (.. request getHeaders (get HttpHeader/CONTENT_TYPE))]
+    (if-let [mime-type-charset (.get MimeTypes/CACHE content-type)]
+      (str mime-type-charset)
+      (MimeTypes/getCharsetFromContentType content-type))))
+
+(defn- get-client-cert [^Request request]
+  ;; TODO: get data from SecureRequestWithTLSData
+  )
+
+
+(defn build-request-map
+  "Create the request map from the Request object."
+  [^Request request]
+  {:server-port        (Request/getLocalPort request)
+   :server-name        (Request/getLocalAddr request)
+   :remote-addr        (Request/getRemoteAddr request)
+   :uri                (when-let [uri (.getHttpURI request)]
```

Comment:
Type hint to avoid reflection ^HttpURI

Reply:
We should not need type hint for getHttpURI because there is no overloading and request is already tagged.